desktop/core/src/desktop/js/apps/storageBrowser/FileChooserModal/FileChooserModal.tsx

```diff
@@ -124,10 +150,68 @@ const FileChooserModal = ({
         if (record.type === BrowserViewType.dir) {
           setDestPath(record.path);
         }
+        if (isFileSelectionAllowed && record.type === BrowserViewType.file) {
+          onSubmit(record.path);
+          onClose();
+        }
       }
     };
   };
 
+  const onFilesDrop = (newFiles: File[]) => {
+    const newUploadItems = newFiles.map(file => {
+      return {
+        file,
+        filePath: destPath,
+        uuid: UUID(),
+        status: FileStatus.Pending
+      };
+    });
+    setPolling(true);
+    huePubSub.publish(FILE_UPLOAD_START_EVENT, {
+      files: newUploadItems
+    });
+  };
+
+  useHuePubSub({
+    topic: FILE_UPLOAD_SUCCESS_EVENT,
+    callback: () => {
+      setPolling(false);
+    }
+  });
+
+  const handleUploadClick = () => {
+    if (!uploadRef || !uploadRef.current) {
+      return;
+    }
+    uploadRef.current.click();
+  };
+
+  const handleFileUpload = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const files = Array.from(e.target.files as ArrayLike<File>);
+    onFilesDrop(files);
+  };
+
+  const {
+    save: createFolder,
+    loading: createFolderLoading,
+    error: createFolderError
+  } = useSaveData(CREATE_DIRECTORY_API_URL, {
+    postOptions: { qsEncodeData: true } // TODO: Remove once API supports RAW JSON payload
+  });
+
+  const handleCreate = () => {
+    createFolder(
+      { path: destPath, name: createFolderValue },
+      {
+        onSuccess: () => {
+          setShowCreateFolderModal(false);
+          setDestPath(prev => `${prev}/${name}`);
```

Comment:
should it be instead, i don't see name defined.

`setDestPath(prev => `${prev}/${createFolderValue}`);`